agent/agent.go

```diff
@@ -21,13 +21,38 @@ import (
 	"github.com/google/gops/signal"
 )
 
-// Start starts the gops agent on a host process. Once agent started,
-// users can use the advanced gops features.
+// Agent represents an agent that enable the advanced gops features
+type Agent struct {
+	// HandleSignals is a boolean that tells whether the agent should listen to
+	// the Interrupt signal and shutdown the applications after performing the
+	// necessary cleanup.
+	HandleSignals bool
+
+	portfile string
+	listener net.Listener
+}
+
+// Start starts the gops agent on a host process. Once agent started, users can
+// use the advanced gops features. The agent will listen to Interrupt signals
+// and exit the process, if you need to perform further work on the Interrupt
+// signal (*Agent).Start() should be used.
 //
 // Note: The agent exposes an endpoint via a TCP connection that can be used by
-// any program on the system. Review your security requirements before
-// starting the agent.
+// any program on the system. Review your security requirements before starting
+// the agent.
 func Start() error {
+	return (&Agent{
```

Comment:
I made a mistake by calling this function Start. In Go, we use Listen/Close. Should we s/Start/Listen and s/Stop/Close while we are here?